src/modules/feature-modules/accessor/pages/innovation/support/organisations-support-status-suggest.component.ts

```diff
@@ -20,109 +33,420 @@ import { SupportLogType } from '@modules/shared/services/innovations.dtos';
+                    unit.id
+                  )
+              )
+            };
+
+            let description = undefined;
+            if (org.organisationUnits.length > 1) {
+              const totalUnits = newOrg.organisationUnits.length;
+              description = `${totalUnits} ${totalUnits > 1 ? 'units' : 'unit'} in this organisation`;
+            }
+
+            return { ...newOrg, description };
+          })
+          .filter(org => org.organisationUnits.length > 0);
+
+        this.organisationItems = this.organisationsToSuggest.map(org => ({
+          value: org.id,
+          label: `${org.name} (${org.acronym})`,
+          description: org.description
+        }));
+
+        this.organisationItems.unshift({ value: 'Select an organisation:', label: 'HEADING' });
+
+        this.setPageStatus('READY');
+      },
+      error: () => {
+        this.setPageStatus('ERROR');
+        this.setAlertUnknownError();
+      }
     });
   }
 
   onSubmitStep(): void {
-    const chosenUnitsValues: string[] = [];
-    const chosenUnitsList = this.groupedItems
-      .map(item => {
-        const units = item.items.filter(
-          i => i.isEditable && (this.form.get('organisationUnits')!.value as string[]).includes(i.value)
-        );
-
-        if (units.length === 0) {
-          return { organisation: '', units: [] };
-        } // This is filtered after the map.
-
-        chosenUnitsValues.push(...units.map(u => u.value));
-        return { organisation: item.label, units: item.items.length !== 1 ? units.map(u => u.label) : [] };
-      })
-      .filter(o => o.organisation);
-
-    this.chosenUnits = { list: chosenUnitsList, values: chosenUnitsValues };
-
-    if (this.chosenUnits.values.length === 0) {
-      this.form.get('organisationUnits')!.setErrors({
-        customError: true,
-        message: 'You need to choose at least one organisation or one unit to suggest'
-      });
-      this.form.get('organisationUnits')!.markAsTouched();
-      return;
+    this.resetAlert();
+
+    switch (this.stepNumber) {
+      case 1:
+        const organisation = this.form.get('organisation');
+        if (!organisation?.value) {
+          this.setAlertError('You have not selected an organisation', {
+            itemsList: [
+              {
+                title: 'Select an organisation and click continue',
+                fieldId: 'organisation1'
+              }
+            ],
+            width: '2.thirds'
+          });
+          organisation?.markAsTouched();
+        } else {
+          const chosenOrganisation = this.organisationsToSuggest.find(org => org.id === organisation?.value!)!;
+
+          this.chosenUnits.organisation = { name: chosenOrganisation.name, acronym: chosenOrganisation.acronym };
+
+          this.chosenUnits.organisation.information = this.organisationsInformation.find(
+            org => org.acronym === chosenOrganisation.acronym
+          );
```

Comment:
This assign could be done at the same time (if you prefer):

```
this.chosenUnits.organisation = {
    name: ...,
    acronym: ...,
    information: ....
}
```

Reply:
Yes! Thank you